Assets/Scripts/Grid/TerrainMod.cs

```diff
@@ -0,0 +1,49 @@
+﻿using UnityEngine;
+using System;
+
+public static class TerrainMod {
+
+	public static float[][] terrainWeight;			// float map of terrain weights for each terrain and unit class
+
+	// Called on app load to generate the terrain weight map
+	public static void CreateWeightMap(){
+		// Initialize the 2d array based on enum lengths
+		terrainWeight = new float[Enum.GetValues(typeof(UnitTypes)).Length][];
+		for(int cnt = 0; cnt < terrainWeight.Length; cnt++) {
+			terrainWeight[cnt] = new float[Enum.GetValues(typeof(TerrainTypes)).Length];
+		}
+
+		// Testing - will get these values from the server
+		// Larger numbers mean more weight. Values of 0 means impassable
```

Comment:
Maybe 999 should be impassable? Looking at logic, remainingRange = range - (What would be 0 - for mountains) = range.  Leaving remaining range > 0